### PR TITLE
add more mpt flag validations

### PR DESCRIPTION
**Pull request by shawnxie999**
### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### Did you update HISTORY.md?

- [ ] Yes
- [ ] No, this change does not impact library users

### CODE REVIEW

**Review by mvadari — COMMENTED** [review]
Don't forget to update the changelog

**Comment by shawnxie999** [comment]
@mvadari don't think that's necessary since the library hasn't released MPT yet and it is just an extra validation

**Review by khancode — APPROVED** [review]
LGTM

File: packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts
Comment (khancode) [inline packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts:162]: Can you move `MAX_TRANSFER_FEE` to top of this module?

**Review by khancode — CHANGES_REQUESTED** [review]

File: packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts
Comment (khancode) [inline packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts:163]: Should add a ValidationError when `TransferFee` isn't a `number`.

File: packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts
Comment (khancode) [inline packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts:166]: Specify the range in the error message.
```suggestion
        'MPTokenIssuanceCreate: TransferFee must be between 0 and ' + MAX_TRANSFER_FEE,
```

File: packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts
Comment (shawnxie999) [reply packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts:163]: don't think that's necessary since `TransferFee` only supports `number` type

File: packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts
Comment (khancode) [inline packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts:117]: Is this applied to the entire module or function below?

Comment (shawnxie999) [reply packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts:117]: just the function below

Comment (khancode) [reply packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts:117]: Ok, move this below the docstring and above the function signature.

File: packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts
Comment (shawnxie999) [reply packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts:166]: updated

File: packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts
Comment (khancode) [inline packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts:165]: I would put the flags code inside the if block. And change the conditional check to 'tx.TransferFee != null' since the number check is done before it.

**Review by khancode — APPROVED** [review]
One more change and it LGTM.

File: packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts
Comment (khancode) [reply packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts]: Ok, it looks confusing because ideally we're checking if the field is not null. Maybe add in != null with the number check?

Comment (khancode) [reply packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts]: Ok, it looks confusing because ideally we're checking if the field is not null. Maybe add in != null with the number check?

Comment (shawnxie999) [reply packages/xrpl/src/models/transactions/MPTokenIssuanceCreate.ts]: adding the check would also give another lint error. The current condition `typeof tx.TransferFee === 'number' ` already implicitly checks that the field is not null

File: packages/xrpl/test/models/MPTokenIssuanceSet.test.ts
Comment (achowdhry-ripple) [inline packages/xrpl/test/models/MPTokenIssuanceSet.test.ts:61]: Instead of redefining every test, might be easier to follow the pattern used in other test files (define the object in the beforeEach runner, and then selectively set missing fields to undefined or adding to the object over each test)

**Review by achowdhry-ripple — CHANGES_REQUESTED** [review]
Test code cleanup but besides that LGTM

**Review by achowdhry-ripple — APPROVED** [review]
Approving preemptively so I don't block the release, but would recommend the small cleanup comment before merging

**Pull request merged**
